### ISSUE TITLE
recursive glob in top-level package follows "bazel-*" symlinks

### ISSUE THREAD

**Issue by laszlocsomor** [bazelbuild/bazel#6350]
### Description of the problem / feature request:

A `glob` in the top-level package will follow the "bazel-*" convenience symlinks and pick up sources from them.

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

1. `git clone https://github.com/laszlocsomor/projects`
2. `cd bazel/examples/glob_in_root_pkg`
3. `bazel build` -- just do a blank build so we have the "bazel-*" symlinks created
4. `bazel build //:x --spawn_strategy=standalone`
5. `cat bazel-genfiles/x.txt`

This builds happily on Windows, and the x.txt output will list not only "a.txt" but also files under "bazel-out".

On Linux, this fails at Step 4, but because the files under "bazel-out" are not inputs of the genrule action. However, it still shows that the glob picked up the files.

### What operating system are you running Bazel on?

Linux / Windows 10

### What's the output of `bazel info release`?

```
release 0.17.2
```

**Comment by laszlocsomor** [comment]
@laurentlb : Do you have any updates on this bug? Is P1 appropriate? Is this something to fix before Bazel 1.0?

**Comment by laurentlb** [comment]
I think it should be triaged.

**Comment by dslomov** [comment]
Who owns `glob()` implementation? team-Starlark or team-Core? cc @laurentlb @ericfelly

**Comment by laszlocsomor** [comment]
I've been working on glob() recently, I could look at this.

**Comment by laszlocsomor** [comment]
This seems to be fixed in Bazel 0.28.1.

**Comment by mcwilson07** [comment]
I believe this has regressed at some point. I can reproduce the problem with the following two files in my top level WORKSPACE directory.

**BUILD.bazel**
```
load("//:print_files.bzl", "print_files")
print_files(name="glob-defect", files_to_print=glob(["**/*"]))
```

**print_files.bzl**
```
def _print_files_impl(ctx):
    print(ctx.label)
    for file_to_print in ctx.attr.files_to_print:
        print(file_to_print)

print_files = rule(
    implementation = _print_files_impl,
    attrs = {
        'files_to_print' : attr.label_list(allow_files = True)
    },
)
```

The first time I run in an new workspace everything looks fine (using the cpp-examples directory to test).
```
~/Code/examples/cpp-tutorial/stage1$ bazel build //...
DEBUG: examples/cpp-tutorial/stage1/print_files.bzl:3:10: //:glob-defect
DEBUG: examples/cpp-tutorial/stage1/print_files.bzl:5:14: <input file target //:BUILD.bazel>
DEBUG: examples/cpp-tutorial/stage1/print_files.bzl:5:14: <input file target //:README.md>
DEBUG: examples/cpp-tutorial/stage1/print_files.bzl:5:14: <input file target //:WORKSPACE>
DEBUG: examples/cpp-tutorial/stage1/print_files.bzl:5:14: <input file target //:print_files.bzl>
INFO: Analyzed 2 targets (1 packages loaded, 5 targets configured).
INFO: Found 2 targets...
INFO: Elapsed time: 0.247s, Critical Path: 0.02s
INFO: 0 processes.
INFO: Build completed successfully, 1 total action
```

The next time I run Bazel has created the symlinks and will include them in the glob.
```
~/Code/examples/cpp-tutorial/stage1$ bazel build //...
DEBUG: examples/cpp-tutorial/stage1/print_files.bzl:3:10: //:glob-defect
DEBUG: examples/cpp-tutorial/stage1/print_files.bzl:5:14: <input file target //:BUILD.bazel>
DEBUG: examples/cpp-tutorial/stage1/print_files.bzl:5:14: <input file target //:README.md>
DEBUG: examples/cpp-tutorial/stage1/print_files.bzl:5:14: <input file target //:WORKSPACE>
DEBUG: examples/cpp-tutorial/stage1/print_files.bzl:5:14: <input file target //:bazel-bin/main/_objs/hello-world/hello-world.pic.d>
DEBUG: examples/cpp-tutorial/stage1/print_files.bzl:5:14: <input file target //:bazel-bin/main/_objs/hello-world/hello-world.pic.o>
DEBUG: examples/cpp-tutorial/stage1/print_files.bzl:5:14: <input file target //:bazel-bin/main/hello-world>
DEBUG: examples/cpp-tutorial/stage1/print_files.bzl:5:14: <input file target //:bazel-bin/main/hello-world-2.params>
DEBUG: examples/cpp-tutorial/stage1/print_files.bzl:5:14: <input file target //:bazel-bin/main/hello-world.runfiles/MANIFEST>
DEBUG: examples/cpp-tutorial/stage1/print_files.bzl:5:14: <input file target //:bazel-bin/main/hello-world.runfiles/__main__/main/hello-world>
DEBUG: examples/cpp-tutorial/stage1/print_files.bzl:5:14: <input file target //:bazel-bin/main/hello-world.runfiles_manifest>
DEBUG: examples/cpp-tutorial/stage1/print_files.bzl:5:14: <input file target //:bazel-out/k8-fastbuild/bin/main/_objs/hello-world/hello-world.pic.d>
DEBUG: examples/cpp-tutorial/stage1/print_files.bzl:5:14: <input file target //:bazel-out/k8-fastbuild/bin/main/_objs/hello-world/hello-world.pic.o>
DEBUG: examples/cpp-tutorial/stage1/print_files.bzl:5:14: <input file target //:bazel-out/k8-fastbuild/bin/main/hello-world>
DEBUG: examples/cpp-tutorial/stage1/print_files.bzl:5:14: <input file target //:bazel-out/k8-fastbuild/bin/main/hello-world-2.params>
DEBUG: examples/cpp-tutorial/stage1/print_files.bzl:5:14: <input file target //:bazel-out/k8-fastbuild/bin/main/hello-world.runfiles/MANIFEST>
DEBUG: examples/cpp-tutorial/stage1/print_files.bzl:5:14: <input file target //:bazel-out/k8-fastbuild/bin/main/hello-world.runfiles/__main__/main/hello-world>
DEBUG: examples/cpp-tutorial/stage1/print_files.bzl:5:14: <input file target //:bazel-out/k8-fastbuild/bin/main/hello-world.runfiles_manifest>
DEBUG: examples/cpp-tutorial/stage1/print_files.bzl:5:14: <input file target //:bazel-out/stable-status.txt>
DEBUG: examples/cpp-tutorial/stage1/print_files.bzl:5:14: <input file target //:bazel-out/volatile-status.txt>
DEBUG: examples/cpp-tutorial/stage1/print_files.bzl:5:14: <input file target //:print_files.bzl>
INFO: Analyzed 2 targets (1 packages loaded, 21 targets configured).
INFO: Found 2 targets...
INFO: Elapsed time: 0.327s, Critical Path: 0.02s
INFO: 0 processes.
INFO: Build completed successfully, 1 total action
```
